workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/flows/distribution/DistributionUtilities.kt

```diff
@@ -62,21 +62,21 @@ fun ServiceHub.addPartyToDistributionList(party: Party, linearId: UniqueIdentifi
 }
 
 @Suspendable
-fun FlowLogic<*>.addToDistributionList(tokens: List<AbstractToken<TokenPointer<*>>>) {
-    tokens.forEach { token ->
-        val tokenType = token.tokenType
+fun FlowLogic<*>.addToDistributionList(tokens: List<AbstractToken>) {
+    tokens.filter { it.tokenType as? TokenPointer<*> != null }.forEach { token ->
```

Comment:
assumption here is that all tokens coming this way have token pointers... we should probably refactor this code a bit as it's a little confusing...! Either way it's all going when we get DDGs.